core/src/types/key/common.rs

```diff
@@ -42,6 +40,48 @@ pub enum PublicKey {
     Secp256k1(secp256k1::PublicKey),
 }
 
+impl Serialize for PublicKey {
+    fn serialize<S>(
+        &self,
+        serializer: S,
+    ) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        // String encoded, because toml doesn't support enums
+        let prefix = match self {
+            PublicKey::Ed25519(_) => "ED25519_PK_PREFIX",
```

Comment:
Could we define constants for "ED25519_PK_PREFIX" and "SECP256K1_PK_PREFIX", as they are used in both serialize/deserialize?

Reply:
Good point. Done, thanks!